### PR TITLE
etcd: use lease ID from etcd session

### PR DESCRIPTION
**Pull request by aanm**
As sesssion contains a leaseID Cilium should use the lease ID provided
    by the session. This way if a session is terminated its lease is also
    terminated.
    
    Also to avoid concurrent problems, the access of leaseID should happen
    at the lock of the mutex

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5606)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: pkg/kvstore/etcd.go [hunk 1]
```diff
@@ -198,34 +205,6 @@ func (e *etcdClient) renewSession() error {
...
-
-	return nil
-}
-
-func (e *etcdClient) renewLease() error {
-	if e.lease != nil {
-		e.client.Revoke(ctx.TODO(), e.lease.ID)
-	}
-
-	lease, err := e.client.Grant(ctx.TODO(), int64(LeaseTTL.Seconds()))
```
Comment (tgraf) [inline pkg/kvstore/etcd.go:211]: It's a great idea to use the standard lease of the session but then we should call `NewSession()` with `func WithTTL(ttl int) SessionOption {` and pass in `LeaseTTL.Seconds()`

File: pkg/kvstore/etcd.go [hunk 2]
```diff
@@ -151,25 +151,23 @@ type etcdClient struct {
 	// is set up in the etcd Client.
 	firstSession chan struct{}
 
-	client *client.Client
```
Comment (tgraf) [inline pkg/kvstore/etcd.go]: Is this clean up critical? It will make backports more difficult. Please split out the non-critical parts.

**Review by tgraf — CHANGES_REQUESTED** [review]
Great work tracking this down.

File: pkg/kvstore/etcd.go (hunk 2 above)
Comment (aanm) [reply pkg/kvstore/etcd.go]: I think so because it was confusing for me to track down which mutex was protecting what.

Comment (ianvernon) [reply pkg/kvstore/etcd.go]: I agree with @tgraf here.

**Comment by aanm** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**